src/codeActions/utils/snippetBuildingUtils.ts

```diff
@@ -347,9 +349,13 @@ export function getAllEntitiesListsSnippet(entityType: string, extension: Extens
  */
 export function getAllChartCardsSnippet(entityType: string, extension: ExtensionStub): string {
   const typeIdx = extension.topology.types.findIndex((type) => type.name === entityType);
-  const entityMetrics = getEntityMetrics(typeIdx, extension);
+  const extensionDatasource = getExtensionDatasource(extension);
+  const entityMetrics = extensionDatasource.length > 0 ?
+    getEntityMetrics(typeIdx, extension) :
```

Comment:
I would like this function - `getEntityMetrics` - to contain all logic on how to deal with things instead of duplication with slight differences as edge cases come up.

Reply:
We're already suffering from some duplication issues and it will only make things more difficult for maintenance and new collaborators.

Another example of duplication is `getAllMetricKeysFromDataSource` + `getMetricsFromDataSource` + `getAllMetricKeysAndValuesFromDataSource` which could all be squashed and have some extra params for the difference in behavior. But this one is for a separate ticket